flytepropeller/pkg/controller/nodes/executor.go

```diff
@@ -298,6 +299,10 @@ func (c *recursiveNodeExecutor) handleDownstream(ctx context.Context, execContex
 				// If the failure policy allows other nodes to continue running, do not exit the loop,
 				// Keep track of the last failed state in the loop since it'll be the one to return.
 				// TODO: If multiple nodes fail (which this mode allows), consolidate/summarize failure states in one.
+				if executableNodeStatusOnComplete != nil {
+					c.nodeExecutor.Clear(executableNodeStatusOnComplete)
```

Comment:
I think it makes sense to add the `enableCRDebugMetadata bool` argument here to the `ClearExecutionError` function on the `MutableNodeStatus` interface. Then this call more closely reflects the `UpdatePhase` call above and we can remove adding a `Clear` function to the `nodeExecutor` struct and similarly the `NodeExecutor` interface. Thoughts?

Reply:
I've never used golang before I started using flyte so I have little opinion on how the interfaces are organised. I will try to implement it as you suggested.